src/map/map.c

```diff
@@ -2109,14 +2109,12 @@ void map_foreachregen(int (*func)(struct block_list* bl, va_list args), ...)
 	iter = db_iterator(regen_db);
 	for( bl = (struct block_list*)dbi_first(iter); dbi_exists(iter); bl = (struct block_list*)dbi_next(iter) )
 	{
-		va_list args;
-		int ret;
-
-		va_start(args, func);
-		ret = func(bl, args);
-		va_end(args);
-		if( ret == -1 )
-			break;// stop iterating
+		// Dead BLs don't regenerate
+		// TODO: BLs other than PC should issue a warning/error since they might refer to bad code(not removed bls)
+		if( !status_isdead(bl) ){
+			if( func(bl) == -1 )
+				break;// stop iterating
```

Comment:
I totally agree with you. I just kept it like it was.
We never ever return -1 anyway, so I guess we should really remove it after all. :)